src/nspawn/nspawn.c

```diff
@@ -2684,6 +2684,10 @@ static int inner_child(
                 }
         }
 
+        r = reset_uid_gid();
```

Comment:
This fixes: https://gist.github.com/evverx/adf78f66cf4933a436f92d8f8d6721e7#file-how_to_run-md